File: wwwroot/js/ProjectClass/SJC_Project.js
import { FramingInvoice } from "./FramingInvoice.js";
import { FormingInvoice } from "./FormingInvoice.js";
export class SJC_Project {
    _id;
    Title;
    IsCompelete;
    ContractDate;
    TotalBudget;
    FramingBudget;
    BuildingQty;
    PercentBase;
    FramingContractNo;
    FormingContractNo;
    FramingTitles;
    FramingInvoiceList;
    FormingInvoiceList;

    constructor() {
        this._id = "";
        this.Title = "";
        this.IsCompelete = false;
        this.ContractDate = "";
        this.TotalBudget = 0;
        this.FramingBudget = 0;
        this.BuildingQty = 0;
        this.PercentBase = false;
        this.FramingContractNo = "";
        this.FormingContractNo = "";
        this.FramingTitles = [""];
        this.FramingTitles.pop();
        this.FormingTitles = [""];
        this.FormingTitles.pop();
        this.FramingInvoiceList = [new FramingInvoice()];
        this.FramingInvoiceList.pop();
        this.FormingInvoiceList = [new FormingInvoice()];
        this.FormingInvoiceList.pop();
    }
}

File: Data/Project Classes/SJC_Project.cs
﻿using MongoDB.Bson;
using SJC_Portal.Data.Project_Classes;

namespace SJC_Portal.Data
{
    public class SJC_Project
    {
        public ObjectId? _id { get; set; }
        public string? Title { get; set; }
        public bool? IsCompelete { get; set; }
        public string? ContractDate { get; set; }
        public double? TotalBudget { get; set; }
        public double? FramingBudget { get; set; }
        public int? BuildingQty { get; set; }
        public bool? PercentBase { get; set; }
        public string? FramingContractNo { get; set; }
        public string? FormingContractNo { get; set; }
        public IEnumerable<string>? FramingTitles { get; set; }
        public IEnumerable<string>? FormingTitles { get; set; }
        public IEnumerable<FramingInvoice>? FramingInvoiceList { get; set; }
        public IEnumerable<FormingInvoice>? FormingInvoiceList { get; set; }
    }
}
